contracts/RewardController.sol

```diff
@@ -152,28 +157,27 @@ contract RewardController is IRewardController, OwnableUpgradeable {
             }
         }
         rewardDebt[_vault][_user] = userShares * rewardPerShare / 1e12;
-        if (pending > 0 && _claimReward) {
-            safeTransferReward(_user, pending, _vault);
-        }
     }
 
-    function updateVaultBalance(
-        address _user,
-        uint256 _sharesChange,
-        bool _isDeposit,
-        bool _claimReward
-    ) external {
-        _updateVaultBalance(msg.sender, _user, _sharesChange, _isDeposit, _claimReward);
+    function updateVaultBalance(address _user, uint256 _sharesChange, bool _isDeposit) external {
```

Comment:
feels a bit wrong to have this public, but I see no attack here so np. Perhaps add a docstring saying that this is meant to be called by the vault, but will be harmless if called by a non-vault address

Reply:
Yeah I felt that too at first, but then since there is no harm or risk in having it open I guess it's cleaner and saves gas compared to adding limitations